common/mariadb-galera/helm/tests/02-services-mariadb-galera_test.yaml

```diff
@@ -974,3 +974,45 @@ tests:
       - equal:
           path: spec.sessionAffinityConfig.clientIP.timeoutSeconds
           value: 28800
+
+  - it: database service name changed if migration option is enabled
+    template: service.yaml
+    documentSelector:
+      path: metadata.name
+      value: testrelease-mariadb-replica
+    set:
+      mariadb.galera.clustername: "testclustername"
+      mariadb.migration.enabled: true
+    asserts:
+      - isKind:
+          of: Service
+      - equal:
+          path: metadata.name
+          value: testrelease-mariadb-replica
+      - notEqual:
+          path: metadata.name
+          value: testrelease-mariadb
+      - equal:
+          path: metadata.labels.app
+          value: testrelease
+
+  - it: database service name changed if migration option is disabled
+    template: service.yaml
+    documentSelector:
```

Comment:
you are already checking for the document with name testrelease-mariadb i would use documentIndex instead 
like this 
https://github.com/sapcc/helm-charts/blob/b2428bb069158e5fc8e4b0378c04b755a8f52659/common/mariadb-galera/helm/tests/02-services-mariadb-galera_test.yaml#L222

Reply:
I have opted against `documentIndex` because it will break if the order of documents (because of new features) will change. My `documentSelector` is somewhat redundant, but it should be more stable...